src/iceberg_functions/iceberg_scan.cpp

```diff
@@ -15,302 +15,870 @@
+               expr.type == ExpressionType::COMPARE_LESSTHANOREQUALTO) {
+        // Clone the expression and add to predicates
+        predicates.emplace_back(expr.Copy());
+    }
+    // Add more conditions here if you want to handle OR or other expressions
+}
+
 struct IcebergScanGlobalTableFunctionState : public GlobalTableFunctionState {
 public:
-	static unique_ptr<GlobalTableFunctionState> Init(ClientContext &context, TableFunctionInitInput &input) {
-		return make_uniq<GlobalTableFunctionState>();
-	}
+    static unique_ptr<GlobalTableFunctionState> Init(ClientContext &context, TableFunctionInitInput &input) {
+        return make_uniq<GlobalTableFunctionState>();
+    }
 };
 
 static unique_ptr<ParsedExpression> GetFilenameExpr(unique_ptr<ColumnRefExpression> colref_expr) {
-	vector<unique_ptr<ParsedExpression>> split_children;
-	split_children.push_back(std::move(colref_expr));
-	split_children.push_back(make_uniq<ConstantExpression>(Value("/")));
-	auto data_split = make_uniq<FunctionExpression>("string_split", std::move(split_children));
+    vector<unique_ptr<ParsedExpression>> split_children;
+    split_children.emplace_back(std::move(colref_expr));
+    split_children.emplace_back(make_uniq<ConstantExpression>(Value("/")));
+    auto data_split = make_uniq<FunctionExpression>("string_split", std::move(split_children));
 
-	vector<unique_ptr<ParsedExpression>> list_extract_children;
-	list_extract_children.push_back(std::move(data_split));
-	list_extract_children.push_back(make_uniq<ConstantExpression>(Value(-1)));
-	auto list_extract_expr = make_uniq<FunctionExpression>("list_extract", std::move(list_extract_children));
+    vector<unique_ptr<ParsedExpression>> list_extract_children;
+    list_extract_children.emplace_back(std::move(data_split));
+    list_extract_children.emplace_back(make_uniq<ConstantExpression>(Value(-1)));
+    auto list_extract_expr = make_uniq<FunctionExpression>("list_extract", std::move(list_extract_children));
 
-	return std::move(list_extract_expr);
+    return std::move(list_extract_expr);
 }
 
 static unique_ptr<ParsedExpression> GetFilenameMatchExpr() {
-	auto data_colref_expr = make_uniq<ColumnRefExpression>("filename", "iceberg_scan_data");
-	auto delete_colref_expr = make_uniq<ColumnRefExpression>("file_path", "iceberg_scan_deletes");
+    auto data_colref_expr = make_uniq<ColumnRefExpression>("filename", "iceberg_scan_data");
+    auto delete_colref_expr = make_uniq<ColumnRefExpression>("file_path", "iceberg_scan_deletes");
 
-	auto data_filename_expr = GetFilenameExpr(std::move(data_colref_expr));
-	auto delete_filename_expr = GetFilenameExpr(std::move(delete_colref_expr));
+    auto data_filename_expr = GetFilenameExpr(std::move(data_colref_expr));
+    auto delete_filename_expr = GetFilenameExpr(std::move(delete_colref_expr));
 
-	return make_uniq<ComparisonExpression>(ExpressionType::COMPARE_NOT_DISTINCT_FROM, std::move(data_filename_expr),
```

Comment:
All these format changes make the code hard to review - can we leave the old format in place?